bundle/service/src/test/java/hu/uni/miskolc/iit/swtest/team3/service/ReaderServiceImplTest.java

```diff
@@ -0,0 +1,146 @@
+
+    @Before
+    public void init() {
+        readerServiceImpl = new ReaderServiceImpl(testBookDao, testBorrowingDao);
+
+        testBook.setIsbn("0-684-84328-5");
+        testBook.setAuthor("Author Arthur");
+        testBook.setTitle("The test");
+        testBook.setDescription("A real good one for testing");
+        testBook.setLanguage("English");
+        testBook.setAvailableCopies(1);
+
+        testBookList.add(testBook);
+
+        testBorrowing.setBorrowId(1);
+        testBorrowing.setStatus(BorrowStatus.RETURNED);
+        testBorrowing.setCreatorId(66);
+        testBorrowing.setBookIsbn("811326746-8");
+        testBorrowing.setCreationDate(TEST_CREATIONDATE);
+
+        testBorrowingList.add(testBorrowing);
+
+        testUser.setUserId(66);
+        testUser.setPasswordHash("$2a$frtF_$dasgew");
+        testUser.setName("Test Thomas");
+        testUser.setEmail("test.thomas88@unittest.org");
+        testUser.setLibrarian(Boolean.TRUE);
+    }
+
+    @Test
+    public void testListBooks() {
+        when(testBookDao.read()).thenReturn(testBookList);
+
+        Assert.assertEquals(testBookList, readerServiceImpl.listBooks());
+        verify(testBookDao).read();
+    }
+
+    @Test(expected = UnsuccessfulOperationException.class)
+    public void testListBooksException() {
+        Mockito.when(testBookDao.read()).thenThrow(Mockito.mock(DataAccessException.class)).thenReturn(testBookList);
+
+        Assert.assertEquals(testBookList, readerServiceImpl.listBooks());
+        verify(testBookDao).read();
+    }
+
+    @Test
+    public void testListBorrowings() {
+        when(testBorrowingDao.readByUser(testUser)).thenReturn(testBorrowingList);
+
+        Assert.assertEquals(testBorrowingList, readerServiceImpl.listBorrowings(testUser));
+        verify(testBorrowingDao).readByUser(testUser);
+    }
+
+    @Test(expected = UnsuccessfulOperationException.class)
+    public void testListBorrowingsException() {
+        Mockito.when(testBorrowingDao.readByUser(testUser)).thenThrow(Mockito.mock(DataAccessException.class)).thenReturn(testBorrowingList);
+
+        Assert.assertEquals(testBorrowingList, readerServiceImpl.listBorrowings(testUser));
+        verify(testBorrowingDao).readByUser(testUser);
+    }
+
+    @Test
+    public void testRequestBook() {
+        when(testBookDao.read(testBook.getIsbn())).thenReturn(testBook);
+        readerServiceImpl.requestBook(testBook, testUser);
+    }
+
+    @Test(expected = UnsuccessfulOperationException.class)
+    public void testRequestBookException() {
+        Mockito.when(testBookDao.read(testBook.getIsbn())).thenThrow(Mockito.mock(DataAccessException.class)).thenReturn(testBook);
+
+        readerServiceImpl.requestBook(testBook, testUser);
+    }
+    
+
+    @Test
+    public void testCheckAvailability(){
+        when(testBookDao.read(testBook.getIsbn())).thenReturn(testBook);
+        
+        if(testBook.getAvailableCopies() > 0){

```

Comment:
How does this work?
How many available copies the "testBook" has
If you want to check both if and else you should do it in separate test or something.
I dont think using if is a good idea here